### PR TITLE
#167765386 Setup docker for backend

### PR DESCRIPTION
**Pull request by victor-nach**
## What does this PR do?
- Setup docker for backend application
## Description of Task to be completed?
- Create a docker file that would create an image to run the application
- Create a docker-compose.yml file that would create and run the app with the required services
## How should this be manually tested?
- clone this branch
- Install docker and docker-compose locally on your system
- run `docker build .` to run Dockerfile that would create the image
- run `docker-compose up` that would run the docker image and create an environment
## Any background context you want to provide?
- ensure that you have a start script in your package.json
- make sure to add relevant environment variables in a .env file (DATABASE_URL, PORT, POSTGRES_USER, POSTGRES_DB)
## What are the relevant pivotal tracker stories?
- #167765386

## Implementation Plan
https://docs.google.com/document/d/1jXBvH_vwOFzvbeL-A-RbPzkH5hQaS0pPbtsLZeuLsu8/edit?ts=5d570bc8

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const app = express();
 app.use(cors());
 
 // Normal express config defaults
-app.use(require("morgan")("dev"));
+// app.use(require("morgan")("dev"));
```
Comment (misanj) [inline src/index.js]: Is there a reason why you commented out this line?

**Review by misanj — APPROVED** [review]
Looks good to me but I suggest you fix the new line error on `dockerfile` and `docker-compose.yml`

**Review by okobaba1 — APPROVED** [review]
Nice one

File: src/index.js (hunk 1 above)
Comment (victor-nach) [reply src/index.js]: the app couldn't start because morgan is not listed in the package.json.

**Review by debelistic — APPROVED** [review]

**Comment by anneKay** [comment]
Please resolve the conflict in this PR

File: package.json [hunk 2]
```diff
@@ -4,10 +4,8 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "start": "node ./src/bin/www",
-    "devstart": "babel-watch ./src/index.js",
-    "build": "rm -rf dist && mkdir dist && babel src -s -d dist",
-   "test": " set NODE_ENV=test&& mocha -r @babel/register --recursive src/tests/ --exit"
+    "start": "node src",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (anneKay) [inline package.json]: Why did you remove the devstart and build script. I also notice you are not using the current version of our test script.

Comment (victor-nach) [reply package.json]: I didn't remove them, in the version I checked out from, that's how it was. Let me just rebase it again

Comment (anneKay) [reply package.json]: If that was the code from the develop branch you checked out from and you didn't touch that code, why is it showing up as your change here. Ii this PR get's merged, what you currently have will override the current state of develop. I think that we all need to go over git, how we rebase with develop, recheck the workflow the team has following.

File: src/index.js (hunk 1 above)
Comment (anneKay) [inline src/index.js]: Why is this line commented?

File: package.json (hunk 2 above)
Comment (victor-nach) [reply package.json]: Okay I see what you're saying. I've resolved it now, thanks.
